reels/fluter/__init__.py

```diff
@@ -0,0 +1,102 @@
+"""
+Module for sending to deep pixelflut server.
+
+Simple functions to send to a pixelflut server from a python script.
+Handles connecting to the server on its own (but does never disconnect).
+In addition to setting individual pixels it supports a command "WL" to
+send a complete picture to our LED-Tetris-Wall.
+
+To specify host/port of target server, set environment variable
+PIXELFLUT_HOST to "hostname" or to "hostname:port".
+
+Created by deep cyber -- the deepest of all cybers.
+"""
+
+import socket
+import base64
+import numpy as np
+import os
+
+HOST = "localhost"
+PORT = 1234
+WIDTH = 16
+HEIGHT = 24
+DEPTH = 3
+
+# connect socket
+sock = None
```

Comment:
IMO it is an antipattern to use global variables for sockets. I get that it is convenient since you don't have to pass the socket around but so is having everything encapsulated in an object.

Why not write a `class Fluter`? This would solve
- the awkward configuration (use init params instead of global variables)
- being able to stream to multiple pixel walls from the same code

For example:
```python
class Fluter:
    def __init__(self, host, width=24, height=16, depth=3):
        self.host, self.port = self._parse_host(host)
        self.width = width
        # [...]
    def _connect(self):
        # use self.host, self.port to connect
    def send_img(self, image):
        # ...
```
For convenience you could also have a static method that does init + connect:
```python
class Fluter:
    @staticmethod
    def connect(cls, host, **kwargs):
        fluter = cls(host, **kwargs)
        fluter._connect()
        return fluter

Reply:
I'm on it. From where I started, this kinda made sense - but looking at it without the history, your points are all sound. Wanted to keep it dead simple - but one class instance should be okay in a python script reel, I guess.

Reply:
done. Is a class now. Just creating an instance and using that works as before, only with methods now. Socket now lives inside object, so multiple connections to different pixelflut servers should be possible. Connection still happens on demand at first send for convenience.